src/gfn/gflownet/flow_matching.py

```diff
@@ -84,6 +84,8 @@ def flow_matching_loss(
             AssertionError: If the batch shape is not linear.
             AssertionError: If any state is at $s_0$.
         """
+        if len(states) == 0:
+            return torch.tensor(0.0, device=states.device)
```

Comment:
nice - this resolves a bug I encountered last week actually. I think this is also something that can happen with all losses that rely on `Transitions` -- if the submitted batch has no valid transitions, the code cannot handle it (this is an edge case I mostly ran into when sampling an unlucky batch from a too-small replay buffer).